### PR TITLE
Add `selected` callback for TextInput 

### LINKED ISSUE #6450: Feature Request: TextInput `selected` callback
It would be nice to have `selected` callback for TextInput, that invokes only once after you released key.
Now there is only `cursor-position-changed` callback, that invokes  multiple times while you selecting text.

### PR DESCRIPTION
**Pull request by lmaxyz**
ChangeLog: Added `selected` callback for TextInput

Closes #6450

### CODE REVIEW

**Comment by ogoffart** [comment]
Thanks for the patch!

This only call the callback if the selection is made with the mouse, and only when releasing the mouse.
I think it should also be called when the selection is done with the keyboard, not?

The new callback should be documented in docs/reference/src/language/builtins/elements.md

A test would be nice. For example, tests/cases/text/select_double_click.slint could be extended with checking that the callback was called.  There are other test in that folder that can be used as inspiration.

**Comment by lmaxyz** [comment]
> This only call the callback if the selection is made with the mouse, and only when releasing the mouse. I think it should also be called when the selection is done with the keyboard, not?

You are right. I'll add these changes.

**Comment by lmaxyz** [comment]
I met some code in tests, which made me think we can make this callback inside `slint` file with properties and callbacks we have now:
```slint
property <bool> text_selected: self.anchor-position-byte-offset != self.cursor-position-byte-offset;
cursor-position-changed(new_pos) => {
    if text_selected {
        ...
    }
}
```
And I'm not sure about callback naming anymore. It's more `selection_ended` or something like this if we invoke it only when mouse key released.
And by keyboard it looks like it should be invoked only when shift key released, but now it's only possible to invoke it per any keyboard move event or "select all" shortcut and I'm not sure it matches with mouse's behavior.

File: internal/core/items/text.rs [hunk 1]
```diff
@@ -595,9 +596,15 @@ impl Item for TextInput {
                 #[cfg(not(target_os = "android"))]
                 self.ensure_focus_and_ime(window_adapter, self_rc);
             }
-            MouseEvent::Released { button: PointerEventButton::Left, .. } => {
+            MouseEvent::Released { position, button: PointerEventButton::Left, .. } => {
+                let released_offset = self.byte_offset_for_position(position, window_adapter) as i32;
                 self.as_ref().pressed.set(0);
                 self.copy_clipboard(window_adapter, Clipboard::SelectionClipboard);
+                
+                if self.anchor_position_byte_offset() != released_offset {
```
Comment (ogoffart) [inline internal/core/items/text.rs:604]: I think this is kind of not the right condition, as it shows in the test. Maybe it should be if there is a selection at all?

**Comment by lmaxyz** [comment]
@ogoffart do you really think this callback need if we already can implement it by this code?
```slint
property <bool> text_selected: self.anchor-position-byte-offset != self.cursor-position-byte-offset;
cursor-position-changed(new_pos) => {
    if text_selected {
        ...
    }
}
```

**Pull request closed without merging**